tests/sieves/test_abi_compat.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+# thoth-adviser
+# Copyright(C) 2020 Kevin Postlethwait
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Test filtering out Python Packages based on required and available ABI symbols."""
+
+import pytest
+import flexmock
+
+from thoth.adviser.exceptions import NotAcceptable
+from thoth.adviser.sieves import AbiCompatabilitySieve
+from thoth.adviser.context import Context
+from thoth.adviser.enums import RecommendationType
+from thoth.python import PackageVersion
+from thoth.python import Source
+from thoth.storages import GraphDatabase
+from thoth.storages.exceptions import NotFoundError
+
+from ..base import AdviserTestCase
+
+system_symbols = ["GLIBC_2.0", "GLIBC_2.1", "GLIBC_2.2", "GLIBC_2.3", "GLIBC_2.4", "GLIBC_2.5", "GCC_3.4", "X_2.21"]
+required_symbols_a = ["GLIBC_2.9"]
+required_symbols_b = ["GLIBC_2.4"]
+
+
+class TestAbiCompatSieve(AdviserTestCase):
+    """Test filtering out packages based on enabled or disabled Python package index."""
+
+    def test_abi_compat_symbols_present(self) -> None:
```

Comment:
You can add `graph` as a parameter to both tests - `graph` is a fixture that gets instantiated and mocked by flexmock for you:

```python
    def test_abi_compat_symbols_present(self, graph: GraphDatabase) -> None:
```

Reply:
This change seems to break the test because `flexmock` tries to connect to the database looking for it on localhost rather than leaving it as a partial class declaration.

Reply:
Right. OK, so leave it as is. :+1: